include/common/coredump.h

```diff
@@ -0,0 +1,287 @@
+      Content.insert(Content.end(), Iter->getData().begin(),
+                     Iter->getData().end());
+    }
+    if (Content.size() == 0) {
+      std::cout << "empty\n";
+      std::cout << Content.max_size();
+      Content.insert(Seg.getData().end(), 0x00);
+    }
+    std::cout << Content.size();
+    Datasec.getContent().push_back(Seg);
+    return Datasec;
+  }
+
+  // corestack   ::= customsec(thread-info vec(frame))
+  // thread-info ::= 0x0 thread-name:name
+  // frame       ::= 0x0 funcidx:u32 codeoffset:u32
+  //                     locals:vec(value) stack:vec(value)
+  AST::CustomSection collectCoreStack(Runtime::StackManager &StackMgr) {
+    spdlog::info("Constructing CoreStack");
+
+    AST::CustomSection CoreStack;
+    CoreStack.setName("corestack");
+
+    auto Frames = StackMgr.getAllFrames();
+
+    // TODO get thread info
+    auto &Content = CoreStack.getContent();
+    for (auto &Item : Frames) {
+      if (Item.Module == nullptr) // guard
+        continue;
+
+      auto Funcidx = Item.From->getSourceIndex(); // function index
+      // TODO which to use?
+      //  auto Funcidx = Item.From->getTargetIndex(); // function index
+      auto Codeoffset = Item.From->getOffset();
+      // auto Codeoffset = 0; // only if codeoffset is unknown?
+      auto LocalsN = Item.Locals;
+
+      auto Locals = StackMgr.getRangeSpan(Item.VPos, LocalsN + Item.VPos);
+      auto Stack = StackMgr.getTopSpan(StackMgr.size() -
+                                       LocalsN); // TODO is this correct?
+
+      Content.push_back(Funcidx);
+      Content.push_back(Codeoffset);
+      //
+      // // TODO get the number type before inserting in the bytes
+      // Is that even required?
+      for (auto &Iter : Locals) {
+        Content.push_back(Iter.unwrap());
+      }
+      for (auto &Iter : Stack) {
+        Content.push_back(Iter.unwrap());
+      }
+    }
+
+    return CoreStack;
+  }
+
+  // AST::GlobalSection collectGlobals(Runtime::StackManager &StackMgr) {
+  //   spdlog::info("Collecting Globals");
+  //   AST::GlobalSection Globals;
+  //   const auto *const CurrentInstance = StackMgr.getModule();
+  //
+  //   auto &GlobalInstances = CurrentInstance->getOwnedGlobalInstances();
+  //   auto &Content = Globals.getContent();
+  //
+  //   for (auto &Iter : GlobalInstances) {
+  //     AST::GlobalSegment Seg;
+  //     auto &Type = Seg.getGlobalType();
+  //     Type = Iter->getGlobalType();
+  //     Content.push_back(Seg);
+  //   }
+  //
+  //   std::cout << Content.size();
+  //
+  //   return Globals;
+  // }
+  // This custom section establishes an index space of modules
+  // that can be referenced in the coreinstances custom section
+  // NOTE This is for multi-module coredumps which are not supported by wasmgdb
+  // as of now
+  // coremodules := customsec(vec(coremodule)) coremodule := 0x0
+  // module-name:name
+  // AST::CustomSection collectCoreModules(Runtime::StackManager &StackMgr) {
```

Comment:
According to the spec,

```
coremodules ::= customsec(vec(coremodule))
coremodule ::= 0x0 module-name:name
```

Even though we support a single module, we should use `getModuleName()` as the `module-name`, not the address of the module.